src/RefractDataStructure.cc

```diff
@@ -143,25 +146,35 @@ namespace drafter {
         throw std::logic_error("Out of scope - ElementFactory for type not implemented");
     }
 
-    static refract::IElement* MsonElementToRefract(const mson::Element& mse);
+    static refract::IElement* MsonElementToRefract(const mson::Element& mse, mson::BaseTypeName defaultNestedType = mson::StringTypeName);
 
-    RefractElements MsonElementsToRefract(const mson::Elements& elements) {
+    RefractElements MsonElementsToRefract(const mson::Elements& elements, mson::BaseTypeName defaultNestedType = mson::StringTypeName) {
         RefractElements result;
-        std::transform(elements.begin(), elements.end(), std::back_inserter(result), MsonElementToRefract);
+        //std::transform(elements.begin(), elements.end(),
+        //               std::back_inserter(result),
+        //               std::bind2nd(std::ptr_fun(MsonElementToRefract), nestedTypeName));
```

Comment:
There is some trouble with `bind2nd` `ptr_fun` and `MsonElementToRefract`, I will check it later.
Temporary it is replaced by loop below.